src/igr-tests/igr_functions

```diff
@@ -0,0 +1,232 @@
+# Included from run-test.sh files.
+# shellcheck disable=SC2034,SC2148 # Disable some false warnings/errors.
+
+### This file contain basic variables for running Integration tests of
+### Dinit.
+
+## Basic functions
+# According to POSIX, echo has some unspecified behavior in some cases, for example
+# when its first argument is "-n" or backslash ("\"). We prefer to rely on
+# POSIX documented things.
+# So we replace shell built-in echo with a printf based function.
+# For more info see http://www.etalabs.net/sh_tricks.html
+echo() {
+    IFS=" " printf %s\\n "$*"
+}
+error() {
+    >&2 echo "${TEST_NAME:-}: Error: $*"
```

Comment:
This uses `"$*"` which will expand all parameters, but then just it specifically checks for a 2nd parameter (`if [ -n "${2:-}" ]; then`). So probably you want to drop the "sub message" part or change `"$*"` to `"$1"`.